src/ra_server_proc.erl

```diff
@@ -634,6 +647,8 @@ follower(_, tick_timeout, State) ->
     {keep_state, State, set_tick_timer(State, [])};
 follower({call, From}, {log_fold, Fun, Term}, State) ->
     fold_log(From, Fun, Term, State);
+follower(_, transfer_leadership, State) ->
+    {next_state, candidate, State, [election_timeout_action(short, State)]};
```

Comment:
It would be quicker to issue the `vote_rpc`s here just like happens when a follower move to pre_vote or a pre_vote moves to candidate. Then the latency of this operation should be shorter. To do so it may be better to move the handling of this message to `ra_server` as that is where that code is currently executed.